experiment/simulation/js/circuit.js

```diff
@@ -0,0 +1,210 @@
+                }
+                else {
+                    listPmos[j].outterminal = -1;
+                    console.log(listPmos[j].outterminal)
+                    console.log("Pmos disabled");
+                    listPmos[j].outvoltage = 0;
+                }
+
+
+            }
+        }
+
+    }
+    for (i = 0; i < listInput.length; i++) {
+        for (j = 0; j < listNmos.length; j++) {
+            const r = listInput[i].id.concat("$", listNmos[j].id);
+            // check if connectionMap have the given nmos input connection
+            if (connectionMap.has(r)) {
+                listNmos[j].midterminal = 1;
+                // if input signal is one
+                if (listInput[i].input === 1) {
+                    if (listNmos[j].voltage === -5) {
+                        listNmos[j].outvoltage = -5;
+                    }
+                    else {
+                        if (listNmos[j].voltage === 0) {
+                            listNmos[j].outvoltage = -9;
+                        }
+                        else {
+                            listNmos[j].outvoltage = 5;
+                        }
+
+                    }
+                    listNmos[j].outterminal = 1;
+
+                    //console.log("Voltage stored in nmos");
+                }
+                else {
+                    listNmos[j].outterminal = 0;
+                    listNmos[j].midterminal = 1;
+                    // listNmos[j].outvoltage = 0;
+                    // console.log("Nmos disabled");
+                }
+
+            }
+            else {  // no carrying of voltage
+                listNmos[j].midterminal = 0;
+
+            }
+        }
+
+    }
+
+    for (i = 0; i < listPmos.length; i++) {
+        for (j = 0; j < listOutput.length; j++) {
+            const r = listPmos[i].id.concat("$", listOutput[j].id);
+
+            if (connectionMap.has(r)) {
+
+                if (listPmos[i].outterminal === 1) {
+                    listOutput[j].voltage = listPmos[i].outvoltage;
+                    pmosNand++;
+                    console.log("Pmos succesfully conducted");
+                    console.log(listOutput[j].voltage)
+                }
+                if (listPmos[i].outterminal === -1) {
+                    listOutput[j].voltage = listPmos[i].outvoltage;
+
+                }
+
+            }
+        }
+
+    }
+    // if any nmos is connected to output then the voltages are propogated based on input signals
+    for (i = 0; i < listNmos.length; i++) {
+        for (j = 0; j < listOutput.length; j++) {
+            const r = listNmos[i].id.concat("$", listOutput[j].id);
+            if (connectionMap.has(r)) {
+                // if nmos 
+                if (listNmos[i].outterminal === 1) {
+                    listOutput[j].voltage = listNmos[i].outvoltage;
+                    nmosNand++;
+                    console.log("Nmos succesfully conducted");
+                }
+
+
```

Comment:
Please use prettier code formatter, and format the javascript (for all experiments pushed)

Reply:
Used it and formatted evrything